### PR TITLE
[REVIEW][FEATURE][GAWE-51][GAWE-67] - Request Reset Password

### PR DESCRIPTION
**Pull request by achmadss**
#### Description

> Request reset password with email and send email containing a link to reset the password
> The link according to @triandamai is `https://gawean.com/{token}`

#### Ticket Reference(s)

[GAWE-51](https://bluehabit.atlassian.net/browse/GAWE-51)

##### UI Changes

> Screenshot or video of UI-related changes

##### Coverage

> Test coverage screenshot.

N/A

#### Note(s)

> Mention other user or extra note unrelated to the MR here.

[GAWE-51]: https://bluehabit.atlassian.net/browse/GAWE-51?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java [hunk 1]
```diff
@@ -62,4 +67,31 @@ public ResponseEntity<BaseResponse<Object>> resetPassword(String token, ResetPas
 
         return BaseResponse.success(translate("auth.success"), new HashMap<>());
     }
+
+    public ResponseEntity<BaseResponse<Object>> requestResetPassword(RequestResetPasswordRequest request) {
+        validate(request);
+        final UserCredential userCredential = userCredentialRepository
```
Comment (triandamai) [inline src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java:73]: kalo pake orElseThrow di bikin functional aja sekalian 

```java
return userCredentialRepository.findByEmail(request.email())
   .map(userCredential->{
     // logic disini
  })
  .orElseThrow(()-> new GeneralErrorException())
```

File: src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java [hunk 2]
```diff
@@ -62,4 +67,31 @@ public ResponseEntity<BaseResponse<Object>> resetPassword(String token, ResetPas
...
+        final UserVerification userVerification = new UserVerification();
+        userVerification.setUser(userCredential);
+        userVerification.setType(VerificationType.RESET);
+        userVerification.setToken(TokenGenerator.generateToken());
+        userVerificationRepository.save(userVerification);
+
+        final boolean isMailed = mailUtil.sendEmail(
+            userCredential.getEmail(),
+            translate("auth.request.reset.password.subject"),
+            "reset-password-request",
```
Comment (triandamai) [inline src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java:86]: dibikin constant aja sih bagusnya

**Review by triandamai — CHANGES_REQUESTED** [review]
cek dolo

File: src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java (hunk 2 above)
Comment (achmadss) [reply src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java:86]: yg dibuat constant apanya ya?

Comment (triandamai) [reply src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java:86]: "reset-password-request"

Comment (triandamai) [reply src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java:86]: nama foldernya

Comment (achmadss) [reply src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java]: check [4519f82](https://github.com/Blue-Habit/gawean-rest-api/pull/12/commits/4519f82341a61216558e6686a40c4a43a7a05a9f)

File: src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java (hunk 1 above)
Comment (achmadss) [reply src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java]: check [ea14393](https://github.com/Blue-Habit/gawean-rest-api/pull/12/commits/ea14393377ef1fc0049af965b5354c17d373f978)

**Review by triandamai — APPROVED** [review]

**Review by triandamai — APPROVED** [review]

File: src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java (hunk 2 above)
Comment (triandamai) [reply src/main/java/com/bluehabit/eureka/services/ResetPasswordService.java]: oke done

**Pull request merged**